src/helpers/getCssExports.ts

```diff
@@ -0,0 +1,147 @@
+import path from 'path';
+import Processor from 'postcss/lib/processor';
+import less from 'less';
+import sass from 'sass';
+import stylus from 'stylus';
+import { CSSExports, extractICSS } from 'icss-utils';
+import { RawSourceMap } from 'source-map-js';
+import tsModule from 'typescript/lib/tsserverlibrary';
+import { createMatchPath } from 'tsconfig-paths';
+import { sassTildeImporter } from '../importers/sassTildeImporter';
+import { Options, CustomRenderer } from '../options';
+import { Logger } from './logger';
+
+export const enum FileType {
+  css = 'css',
+  less = 'less',
+  sass = 'sass',
+  scss = 'scss',
+  styl = 'styl',
+}
+
+export const getFileType = (fileName: string): FileType => {
+  if (fileName.endsWith('.css')) return FileType.css;
+  if (fileName.endsWith('.less')) return FileType.less;
+  if (fileName.endsWith('.sass')) return FileType.sass;
+  if (fileName.endsWith('.styl')) return FileType.styl;
+  return FileType.scss;
+};
+
+const getFilePath = (fileName: string) => path.dirname(fileName);
+
+export interface CSSExportsWithSourceMap {
+  classes: CSSExports;
+  css?: string;
+  sourceMap?: RawSourceMap;
+}
+
+export const getCssExports = ({
+  css,
+  fileName,
+  logger,
+  options,
+  processor,
+  compilerOptions,
+}: {
+  css: string;
+  fileName: string;
+  logger: Logger;
+  options: Options;
+  processor: Processor;
+  compilerOptions: tsModule.CompilerOptions;
+}): CSSExportsWithSourceMap => {
+  try {
+    const fileType = getFileType(fileName);
+    const rendererOptions = options.rendererOptions || {};
+
+    let transformedCss = '';
+    let sourceMap: string | undefined;
+
+    if (options.customRenderer) {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const customRenderer = require(options.customRenderer) as CustomRenderer;
+      transformedCss = customRenderer(css, {
+        fileName,
+        logger,
+        compilerOptions,
+      });
+    } else {
+      switch (fileType) {
+        case FileType.less:
+          less.render(
```

Comment:
What do you think of breaking this function down into smaller helpers?
I have something like this in mind, e.g.

const getCustomRenderer = (...) => [transformedCss, ]
const lessRenderer = (...) => [transformedCss, ]
const sassRenderer = (...) => [transformedCss, sourceMap]
const getRenderer = (fileType) => renderer

const [transformedCss, sourceMap] = getRenderer(fileType)(...)

It's not very important though, so we can change it later